File: app/src/main/java/com/example/seki/android_custom_linklifytext_sample/ClickableTextFragment.java
package com.example.seki.android_custom_linklifytext_sample;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.text.Spannable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class ClickableTextFragment extends Fragment implements ClickableTextPresenter.View {

    ClickableTextPresenter mPresenter;

    EditText mClickableStrings;
    EditText mText;
    Button mSubmitButton;

    RecyclerListAdapter mListAdapter;

    public static ClickableTextFragment getInstance() {
        return new ClickableTextFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState) {
        final View rootView = inflater.inflate(R.layout.fragment_clickable_text, parent, false);
        initRecyclerView(rootView);

        mClickableStrings = (EditText) rootView.findViewById(R.id.clickable_strings);
        mText = (EditText)rootView.findViewById(R.id.text);
        mSubmitButton = (Button)rootView.findViewById(R.id.submit_button);

        mSubmitButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final String clickableStrings = mClickableStrings.getText().toString();
                final String text = mText.getText().toString();
                mPresenter.onClickSubmit(clickableStrings, text);
            }
        });


        return rootView;
    }

    private void initRecyclerView(View rootView) {
        final LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());

        final RecyclerView recyclerView = (RecyclerView)rootView.findViewById(R.id.recycler_list);
        recyclerView.setLayoutManager(layoutManager);

        mListAdapter = new RecyclerListAdapter();
        recyclerView.setAdapter(mListAdapter);
        recyclerView.setHasFixedSize(false);

        final DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(recyclerView.getContext(), layoutManager.getOrientation());
        recyclerView.addItemDecoration(dividerItemDecoration);
    }

    @Override
    public void setPresenter(ClickableTextPresenter presenter) {
        mPresenter = presenter;
    }

    @Override
    public void addData(Spannable text) {
        final ListData data = new ListData(text);
        mListAdapter.addData(data);
    }

    @Override
    public void clearInputText() {
        mClickableStrings.getText().clear();
        mText.getText().clear();
    }

    @Override
    public void showTextEmptyMessage() {
        Toast.makeText(getActivity(), "please input text", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/seki/androidCustomLinklifytextSample/ClickableTextFragment.kt
package com.example.seki.androidCustomLinklifytextSample;

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.text.Spannable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class ClickableTextFragment: Fragment(), ClickableTextPresenter.View {

    private var mPresenter: ClickableTextPresenter? = null

    private var mClickableStrings: EditText? = null
    private var mText: EditText? = null
    private var mSubmitButton: Button? = null

    private var mListAdapter: RecyclerListAdapter? = null

    companion object {
        fun getInstance() = ClickableTextFragment()
    }

    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?, savedInstanceState: Bundle?): View {
        val rootView = inflater.inflate(R.layout.fragment_clickable_text, parent, false)
        initRecyclerView(rootView)

        mClickableStrings = rootView.findViewById(R.id.clickable_strings)
        mText = rootView.findViewById(R.id.text)
        mSubmitButton = rootView.findViewById(R.id.submit_button)

        mSubmitButton!!.setOnClickListener({_ ->
                val clickableStrings = mClickableStrings!!.text.toString()
                val text = mText!!.text.toString()
                mPresenter!!.onClickSubmit(clickableStrings, text)
        })


        return rootView
    }

    private fun initRecyclerView(rootView: View) {
        val layoutManager = LinearLayoutManager(activity)

        val recyclerView = rootView.findViewById<RecyclerView>(R.id.recycler_list)
        recyclerView.layoutManager = layoutManager

        mListAdapter = RecyclerListAdapter()
        recyclerView.adapter = mListAdapter
        recyclerView.setHasFixedSize(false)

        val dividerItemDecoration = DividerItemDecoration(recyclerView.context, layoutManager.orientation)
        recyclerView.addItemDecoration(dividerItemDecoration)
    }

    override fun setPresenter(presenter: ClickableTextPresenter) {
        mPresenter = presenter
    }

    override fun addData(text: Spannable) {
        val data = ListData(text)
        mListAdapter!!.addData(data)
    }

    override fun clearInputText() {
        mClickableStrings!!.text.clear()
        mText!!.text.clear()
    }

    override fun showTextEmptyMessage() {
        Toast.makeText(activity, "please input text", Toast.LENGTH_SHORT).show()
    }
}
